flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriterTest.java

```diff
@@ -249,36 +249,50 @@ public void testRetryableErrorsDoNotViolateAtLeastOnceSemanticsDueToRequeueOfFai
         /*
          * Writing 955 to the sink increases the buffer to size 3 containing [75, 95, 955]. This
          * triggers the outstanding in flight request with the failed 965 to be run, and 965 is
-         * placed at the front of the queue. The first {@code maxBatchSize = 3} elements are
-         * persisted, with 965 succeeding this (second) time. 955 remains in the buffer.
+         * placed at the front of the queue. The failure throttles down {@code maxBatchSize} to 1.
+         * buffer now should be [965, 75, 95, 955]
+         * A new batch containing 965 is then sent, success causes {@code maxBatchSize} to go up
+         * to 3 again.
+         * next batch is then created of all requests, 75 and 95 are also persisted.
+         * 955 is in flight after failure.
          */
         writeXToSinkAssertDestinationIsInStateYAndBufferHasZ(
-                sink, "955", Arrays.asList(25, 55, 965, 75, 95), Arrays.asList(955));
+                sink, "955", Arrays.asList(25, 55, 965, 75, 95), Arrays.asList());
 
         writeXToSinkAssertDestinationIsInStateYAndBufferHasZ(
-                sink, "550", Arrays.asList(25, 55, 965, 75, 95), Arrays.asList(955, 550));
+                sink, "550", Arrays.asList(25, 55, 965, 75, 95), Arrays.asList(550));
```

Comment:
Testing was move comprehensive in the previous PR. I expect this is due to it being easier, since the AIMD is in a separate class. Suggest you reuse the class and tests from previous PR as mentioned in another comment